From 24abd2c1926259be239e993910461627badb091f Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Thu, 8 Nov 2018 16:02:40 +0200
Subject: [PATCH 5077/5242] MLK-20189-4: ASoC: fsl: library_load: Check return
 value for kernel_read

commit  56efef908fd1bb2fd15892f77f6fe5cb12902a91 from
https://source.codeaurora.org/external/imx/linux-imx.git

Bail out if kernel_read returns an error.

Fixes: CID1477415

Reviewed-by: S.j. Wang <shengjiu.wang@nxp.com>
Reviewed-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_dsp_library_load.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_dsp_library_load.c b/sound/soc/fsl/fsl_dsp_library_load.c
index 11bb939..d273f72 100644
--- a/sound/soc/fsl/fsl_dsp_library_load.c
+++ b/sound/soc/fsl/fsl_dsp_library_load.c
@@ -484,7 +484,9 @@ static int get_dyn_info(Elf32_Ehdr *eheader,
 		return -ENOMEM;
 
 	vfs_llseek(file, 0, SEEK_SET);
-	kernel_read(file, srambuf, filesize, &pos);
+	ret_val = kernel_read(file, srambuf, filesize, &pos);
+	if (ret_val < 0)
+		return ret_val;
 	filp_close(file, NULL);
 
 	ret_val = xtlib_split_pi_library_size(
-- 
1.7.9.5

